components/service/fretboard/src/main/java/mozilla/components/service/fretboard/source/kinto/SignatureVerifier.kt

```diff
@@ -0,0 +1,192 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fretboard.source.kinto
+
+import android.util.Base64
+import mozilla.components.service.fretboard.Experiment
+import mozilla.components.service.fretboard.ExperimentDownloadException
+import mozilla.components.service.fretboard.JSONExperimentParser
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.StringReader
+import java.math.BigInteger
+import java.net.URL
+import java.nio.charset.StandardCharsets
+import java.security.InvalidKeyException
+import java.security.NoSuchAlgorithmException
+import java.security.NoSuchProviderException
+import java.security.PublicKey
+import java.security.Signature
+import java.security.SignatureException
+import java.security.cert.CertificateException
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+internal class SignatureVerifier(
```

Comment:
Can you add some comments here describing what the class does? Maybe even add some comments to the code. If you have links to pages that explain more then add them too. :)

Reply:
Okay :)